vms/platformvm/warp/README.md

```diff
@@ -0,0 +1,121 @@
+
+AWM utilizes the weighted validator sets registered BLS Public Keys to each validator to provide a weighted set of BLS Public Keys to verify a messaage coming from any Subnet with Warp Messaging.
+
+BLS Multi-Signatures with Public-Key Aggregation provides a way to aggregate BLS private keys, public keys, and signatures into a single key/signature.
+
+For more information on the cryptographic primitive underlying Warp, see [here](https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html).
+
+## AWM Serialization
+
+Unsigned Message:
+```
++-----------------+----------+--------------------------+
+|      network_id :  uint32  |                 4 bytes  |
++-----------------+----------+--------------------------+
+| source_chain_id : [32]byte |                32 bytes  |
++-----------------+----------+--------------------------+
+|         payload :   []byte |       4 + size(payload)  |
++-----------------+----------+--------------------------+
+                             |  38 + size(payload) bytes|
+                             +--------------------------+
+```
+
+- `networkID` provides replay protection for BLS Signers across different Avalanche Networks
+- `sourceChainID` ensures that each blockchain on Avalanche can only sign a message with its own `blockchainID`
+- `payload` is a byte array containing the contents of the message. VMs can define their own message types to include in the `payload`
+
+Note: the `blockchainID` is the hash of the transaction that created the blockchain on the Avalanche P-Chain. It serves as the unique identifier for the blockchain across the Avalanche Network.
+
+BitSet Signature:
+```
++-----------+----------+---------------------------+
+|   signers :   []byte |          4 + len(signers) |
++-----------+----------+---------------------------+
+| signature : [96]byte |                  96 bytes |
++-----------+----------+---------------------------+
+                       | 100 + size(signers) bytes |
+                       +---------------------------+
+```
+
+- `signers` encodes a BitSet of which validators' signatures are included
+- `signature` is an aggregated BLS Multi-Signature of the Unsigned Message
+
+BitSet Signatures are verified within the context of a specific P-Chain height. At a specific P-Chain height, the PlatformVM serves a canonically ordered validator set for the source subnet. The signers BitSet encodes which validators' signature was included. A 1 at index i in `signers` claims that a signature from the validator at index i in the canonical validator set is included in the aggregate signature.
```

Comment:
I think we should move these definitions of `signers` and `signature` to their declarations above

Reply:
`A 1 at index i in `signers` claims that a signature from the validator at index i in the canonical validator set is included in the aggregate signature.`

We should define how "canonical" is determined

Reply:
Generally, I prefer a short description of the variable and then an elaboration of how it's used below.